src/pow.cpp

```diff
@@ -122,8 +122,14 @@ bool PermittedDifficultyTransition(const Consensus::Params& params, int64_t heig
     return true;
 }
 
+std::function<bool(uint256, unsigned int, const Consensus::Params&)> g_check_pow_mock = nullptr;
+
 bool CheckProofOfWork(uint256 hash, unsigned int nBits, const Consensus::Params& params)
 {
+    if (g_check_pow_mock) {
```

Comment:
Should probably have some belt-and-suspenders check that we're 100% not running on mainnet, maybe based on the contents of `params` and whatever applicable global.

Reply:
An alternative to the mocking would be to use [ `FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION`](https://llvm.org/docs/LibFuzzer.html#fuzzer-friendly-build-mode) in all places that call `CheckProofOfWork` (i.e. validation, mining) since there is literally zero benefit in having this be a blocker for fuzzing.

Reply:
https://github.com/dergoegge/bitcoin/commit/48f82613eb5079f4bca4bedad32077cf534a0a92

Reply:
Opened https://github.com/bitcoin/bitcoin/pull/29305 to discuss, I think it actually makes more sense than the mocking